### PR TITLE
fix(argo-cd): Restart argocd-server when argocd-cm CM is updated

### PR DESCRIPTION
**Pull request by yu-croco**
Resolves https://github.com/argoproj/argo-helm/issues/2455
Resolves https://github.com/argoproj/argo-helm/issues/2450

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/templates/argocd-server/deployment.yaml [hunk 1]
```diff
@@ -27,6 +27,9 @@ spec:
     metadata:
       annotations:
         checksum/cmd-params: {{ include (print $.Template.BasePath "/argocd-configs/argocd-cmd-params-cm.yaml") . | sha256sum }}
+        {{- if (hasKey .Values.server "configEnabled") | ternary .Values.server.configEnabled .Values.configs.cm.create }}
+        checksum/cm: {{ include (print $.Template.BasePath "/argocd-configs/argocd-cm.yaml") . | sha256sum }}
+        {{- end }}
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:32]: same as here https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-application-controller/statefulset.yaml#L26-L28 . 📝

**Comment by yu-croco** [comment]
CI fails, so I raised issue https://github.com/argoproj/argo-helm/issues/2460 . ✏️

**Review by jmeridth — APPROVED** [review]

**Comment by pdrastil** [comment]
@yu-croco @jmeridth I got chat with @crenshaw-dev - this is more workaround as we might get issues opened for dropped API requests on server restart. And yes not all items in CM are reloaded / watched on change so this is a bug for upstream.

**Comment by jmeridth** [comment]
@pdrastil should we revert?

**Comment by pdrastil** [comment]
@jmeridth Good question - you don't change config every 5 minutes so it can help few people for now and I would revert with proper fix in upstream.

**Pull request merged**